lib/mail/message.rb

```diff
@@ -1815,11 +1815,15 @@ def encode!
     # all headers, attachments, etc.  This is an encoded email in US-ASCII,
     # so it is able to be directly sent to an email server.
     def encoded
-      ready_to_send!
-      buffer = header.encoded
-      buffer << "\r\n"
-      buffer << body.encoded(content_transfer_encoding)
-      buffer
+      if @buffer
+        @buffer
+      else
+        ready_to_send!
+        @buffer = header.encoded
+        @buffer << "\r\n"
+        @buffer << body.encoded(content_transfer_encoding)
+        @buffer
+      end
```

Comment:
Rather than memoize the encoded message (which means any change to the message won't be reflected in subsequence `#encoded` calls), let's track down what's mutating your message body.

Could you debug further to identify? Perhaps you're getting double transfer decoding?